openfisca_aotearoa/variables/acts/social_security/disability_allowance/disability_allowance.py

```diff
@@ -0,0 +1,220 @@
+
+
+class disability_allowance__income_limit_clause_12(variables.Variable):
+    value_type = bool
+    default_value = False
+    entity = entities.Person
+    definition_period = periods.WEEK
+    label = "Does the person earn below the Disability Allowance Income Limit as per clause 12?"
+    reference = "https://www.legislation.govt.nz/act/public/2018/0032/latest/DLM6784890.html"
+
+    def formula_2018_11_26(persons, period, parameters):
+        in_relationship = persons("social_security__in_a_relationship", period)
+        income_within_limit = persons("disability_allowance__family_income", period) <= parameters(period).disability_allowance.income_limits.clauses["clause_12"]
+        return in_relationship * income_within_limit
+
+
+class disability_allowance__income_limit_clause_13(variables.Variable):
+    value_type = bool
+    default_value = False
+    entity = entities.Person
+    definition_period = periods.WEEK
+    label = "Does the person earn below the Disability Allowance Income Limit as per clause 13?"
+    reference = "https://www.legislation.govt.nz/act/public/2018/0032/latest/DLM6784890.html"
+
+    def formula_2018_11_26(persons, period, parameters):
+        sole_parent = persons("social_security__sole_parent", period)
+        only_one_child = (persons("social_security__dependent_children", period) == 1)
+        income_within_limit = persons("social_security__income", period) <= parameters(period).disability_allowance.income_limits.clauses["clause_13"]
+        return income_within_limit * only_one_child * sole_parent
+
+
+class disability_allowance__income_limit_clause_14(variables.Variable):
+    value_type = bool
+    default_value = False
+    entity = entities.Person
+    definition_period = periods.WEEK
+    label = "Does the person earn below the Disability Allowance Income Limit as per clause 14?"
+    reference = "https://www.legislation.govt.nz/act/public/2018/0032/latest/DLM6784890.html"
+
+    def formula_2018_11_26(persons, period, parameters):
+        sole_parent = persons("social_security__sole_parent", period)
+        more_than_one_child = (persons("social_security__dependent_children", period) > 1)
+        income_within_limit = persons("social_security__income", period) <= parameters(period).disability_allowance.income_limits.clauses["clause_14"]
+        return more_than_one_child * sole_parent * income_within_limit
+
+
+class disability_allowance__family_income(variables.Variable):
+    value_type = float
+    entity = entities.Person
+    definition_period = periods.WEEK
+    label = "How much does the family earn per week?"
+    reference = "https://www.legislation.govt.nz/act/public/2018/0032/latest/DLM6784890.html"
+
+    def formula(people, period, parameters):
+        family_income = people.family.sum(people.family.members("social_security__income", period), role=entities.Family.PARTNER) + \
```

Comment:
Please remember to _allocate_ all members of the family to a role. Otherwise openfisca will silently fail to assign roles —this should be fixed in the future.